compiler/src/iree/compiler/Codegen/Common/GPU/GPUReduceBankConflicts.cpp

```diff
@@ -28,6 +28,13 @@ static void padAlloc(MLIRContext *context, memref::AllocOp allocOp,
   int64_t innerDim = allocOpShape.back();
   if (ShapedType::isDynamic(innerDim))
     return;
+
+  // Return if we have collapseShape op as an user as padding in that case is
+  // unsupported.
+  for (auto user : allocOp->getUsers()) {
+    if (isa<memref::CollapseShapeOp>(user))
+      return;
+  }
```

Comment:
This needs to look through expand_shape and subview ops as well (something like the following).

```
std::deque<Operation *> users = allocOp->getUsers();
for (auto user : users) {
  if (isa<memref::CollapseShapeOp>(user)) {
    return;
  }
  if (isa<ViewLikeOpInterface>(user)) {
    for (auto u : user.getViewResult().getUsers()) {
      users.push_back(u);
    }
  }
}
```

Reply:
+1